plugins/workspace-plugin/src/workspace-folder-updater.ts

```diff
@@ -13,7 +13,7 @@ export class WorkspaceFolderUpdater {
   private pendingFolders: string[] = [];
   private addingWorkspaceFolderPromise: Promise<void> | undefined;
 
-  constructor(protected readonly updateWorkspaceFolderTimeout = 3000) {}
+  constructor(protected readonly updateWorkspaceFolderTimeout = 10000) {}
```

Comment:
It feels strange that we need 10 seconds here. Do we know where the time goes? Maybe something to investigate some time later. But +1 on the longer timeout.

Reply:
Within testing my PR on the `dogfooding` instance I found that sometimes 3 seconds is not enough.
I guess that 5 seconds - is good value for the `dogfooding` instance, but I'm not sure it will be OK for another instance, for example for `minikube` that running locally on a user machine with low resources.

So, I can reduce that value to 5-7 seconds within current PR and we could see if there is reported issues about it in further. 
WDYT?

At the moment I see that it takes 3-4 seconds on the `dogfooding` instance for:
- using API `che-theia` from `workspace-plugin` send a request on `theia` side to add a project as a workspace folder
- `theia` does some actions and send an event that workspace folders were updated
-  `workspace-plugin` catch the event. 

I'm going to create an issue to investigate why it takes 3-4 seconds and if we can reduce such delays.

Reply:
Having a large timeout is fine: most of the time, it will not be reached.

Reply:
Created the issue for the delay investigation https://github.com/eclipse/che/issues/19566